src/git/zcl_abapgit_git_commit.clas.abap

```diff
@@ -0,0 +1,257 @@
+"! <p class="shorttext synchronized" lang="en">Git Commit</p>
+CLASS zcl_abapgit_git_commit DEFINITION
+  PUBLIC
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+    CLASS-METHODS get_by_branch
+      IMPORTING
+        !iv_branch_name   TYPE string
+        !iv_repo_name     TYPE string
```

Comment:
seen with git eyes, a repository name does not exist, its an URL endpoint containing git objects

in the code, `iv_repo_name` is only used for a progress indicator, suggest using the URL for the progress indicator instead